### PR TITLE
[Java:vertx] Initialize router in init method and re-use router member to create S…

### PR DESCRIPTION
**Pull request by ccozzolino**
…waggerRouter

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Modify MainApiVerticle.mustache template to properly initialize main router and re-use the router to create SwaggerRouter - fix for #7230

### CODE REVIEW

**Comment by ccozzolino** [comment]
No technical committee member is listed for Vertx, tagging @phiz71 since he committed initial Vertx templates.

**Comment by ccozzolino** [comment]
Originally copied @phiz71 since he was initial contributor of Vertx support, also copying larger Java technical committee: @bbdouglas (2017/07) @JFCote (2017/08) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09)

**Comment by ccozzolino** [comment]
@wing328 - any recommendation on a proper resource to review these changes?

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by ccozzolino** [comment]
@wing328 - I followed the instructions at https://help.github.com/articles/changing-author-info/ but it definitely changed way more than I intended. I'm happy to close this PR and open a new one if that is easier.

**Comment by ccozzolino** [comment]
@wing328 - Thanks for catching that my account was not linked. I've adjusted my PR and the account should now be linked correctly.

**Comment by wing328** [comment]
That's the error: 
```
Warning: Permanently added 'github.com,192.30.253.113' (RSA) to the list of known hosts.

fatal: Could not parse object 'f767d456131c29e558e660593b26159ce2d6a844'.

Could not find commit f767d456131c29e558e660593b26159ce2d6a844 in the repo
```

File: modules/swagger-codegen/src/main/resources/JavaVertXServer/MainApiVerticle.mustache [hunk 1]
```diff
@@ -9,26 +9,34 @@ import com.github.phiz71.vertx.swagger.router.SwaggerRouter;
...
 public class MainApiVerticle extends AbstractVerticle {
     final static Logger LOGGER = LoggerFactory.getLogger(MainApiVerticle.class); 
     
-    final Router router = Router.router(vertx);
-    
+    protected Router router;
+
+    @Override
+    public void init(Vertx vertx, Context context) {
+      super.init(vertx, context);
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/JavaVertXServer/MainApiVerticle.mustache]: Align with 4 spaces

File: modules/swagger-codegen/src/main/resources/JavaVertXServer/MainApiVerticle.mustache [hunk 2]
```diff
@@ -9,26 +9,34 @@ import com.github.phiz71.vertx.swagger.router.SwaggerRouter;
...
+      router = Router.router(vertx);
+    }
+
     @Override
     public void start(Future<Void> startFuture) throws Exception {
         Json.mapper.registerModule(new JavaTimeModule());
     	FileSystem vertxFileSystem = vertx.fileSystem();
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/JavaVertXServer/MainApiVerticle.mustache]: Can you realign this ?

**Comment by ccozzolino** [comment]
@wing328 - looks like circleci issue is resolved. Thanks!

**Comment by ccozzolino** [comment]
@wing328 - are we okay to merge this now?

**Comment by wing328** [comment]
@ccozzolino thanks for the PR. 

Does this PR need to urgently go into 2.3.1 which will be released next week, or it's ok to wait till 2.4.0 release?

**Comment by ccozzolino** [comment]
@wing328 - I'd really like to get this into 2.3.1; however, if that complicates your release schedule, then delaying it to 2.4.0 is not a problem.

Thanks again for all your work on this.

**Comment by wing328** [comment]
@ccozzolino thanks for the PR, which has been merged into master (2.3.1).

I need some help from you to look into the following errors:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project swagger-java-vertx-server: Compilation failure: Compilation failure:
[ERROR] /private/tmp/swagger-codegen/samples/server/petstore/java-vertx/async/src/main/java/io/swagger/server/api/verticle/UserApiVerticle.java:[30,27] cannot find symbol
[ERROR] symbol:   class UserApiImpl
[ERROR] location: class io.swagger.server.api.verticle.UserApiVerticle
[ERROR] /private/tmp/swagger-codegen/samples/server/petstore/java-vertx/async/src/main/java/io/swagger/server/api/verticle/StoreApiVerticle.java:[26,28] cannot find symbol
[ERROR] symbol:   class StoreApiImpl
[ERROR] location: class io.swagger.server.api.verticle.StoreApiVerticle
[ERROR] /private/tmp/swagger-codegen/samples/server/petstore/java-vertx/async/src/main/java/io/swagger/server/api/verticle/PetApiVerticle.java:[32,26] cannot find symbol
[ERROR] symbol:   class PetApiImpl
[ERROR] location: class io.swagger.server.api.verticle.PetApiVerticle
[ERROR] -> [Help 1]
[ERROR] 
```
I got these when running "mvn test" under "samples/server/petstore/java-vertx/async"

**Comment by ccozzolino** [comment]
@wing328 - I see the issue. This looks like an issue unrelated to my changes. I incorrectly assumed that the server build ran the individual sample tests which is why I didn't catch this. I'll fix it and submit a new PR.

**Comment by wing328** [comment]
@ccozzolino yup, the issue has nothing to do with your change as it was present in master before merging this PR. Thanks for offering help to fix this.

**Comment by ccozzolino** [comment]
"mvn test" issues addressed in #7359

**Pull request merged**